packages/contracts/test/amo/AaveAMO.t.sol

```diff
@@ -0,0 +1,331 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import "forge-std/Test.sol";
+import {DiamondTestSetup} from "../diamond/DiamondTestSetup.sol";
+import {UbiquityAMOMinter} from "../../src/dollar/core/UbiquityAMOMinter.sol";
+import {AaveAMO} from "../../src/dollar/amo/AaveAMO.sol";
+import {MockERC20} from "../../../../src/dollar/mocks/MockERC20.sol";
+import {MockChainLinkFeed} from "../../src/dollar/mocks/MockChainLinkFeed.sol";
+import {IPool} from "@aavev3-core/contracts/interfaces/IPool.sol";
+import {IAToken} from "@aavev3-core/contracts/interfaces/IAToken.sol";
+import {IVariableDebtToken} from "@aavev3-core/contracts/interfaces/IVariableDebtToken.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract AaveAMOTest is DiamondTestSetup {
+    UbiquityAMOMinter amoMinter;
+    AaveAMO aaveAMO;
+    address collateralOwner =
+        address(0xC959483DBa39aa9E78757139af0e9a2EDEb3f42D); // Aave Sepolia Faucet
+    MockERC20 collateralToken =
+        MockERC20(0xFF34B3d4Aee8ddCd6F9AFFFB6Fe49bD371b8a357); // DAI-TestnetMintableERC20-Aave Sepolia
+    MockChainLinkFeed collateralTokenPriceFeed =
+        MockChainLinkFeed(0x9aF11c35c5d3Ae182C0050438972aac4376f9516); // DAI-TestnetPriceAggregator-Aave Sepolia
+    IAToken aToken = IAToken(0x29598b72eb5CeBd806C5dCD549490FdA35B13cD8); // DAI-AToken-Aave Sepolia
+    IVariableDebtToken vToken =
+        IVariableDebtToken(0x22675C506A8FC26447aFFfa33640f6af5d4D4cF0); // DAI-VariableDebtToken-Aave Sepolia
+    ERC20 AAVE = ERC20(0x88541670E55cC00bEEFD87eB59EDd1b7C511AC9a); // AAVE Token
+
+    // Constants for the test
+    address constant newAMOMinterAddress = address(5); // mock new AMO minter address
+    address constant nonAMO = address(9999); // Address representing a non-AMO entity
+    uint256 constant interestRateMode = 2; // Variable interest rate mode in Aave
+
+    // Mocking the Aave Pool
+    IPool private constant aave_pool =
+        IPool(0x6Ae43d3271ff6888e7Fc43Fd7321a503ff738951); // Aave V3 Sepolia Pool
+
+    function setUp() public override {
+        vm.createSelectFork(vm.rpcUrl("sepolia"));
```

Comment:
The whole codebase uses plain unit tests. We used to have tests run in a forked forge instance but:
1. They tend to fail because of RPC errors
2. They run longer compared to unit tests
3. It's more clear to have everything deployed from scratch in your tests compared to using already deployed contracts which is kind of a "black box"

Can we deploy aave contracts ourselves and use plain unit tests?

Reply:
I understand where you come from. This is a speed/quality balance, will try to match this pattern - however deploying Aave is not that trivial.

Reply:
Couldn't our rpc manager help with the fail in requests problem?

Reply:
> I understand where you come from. This is a speed/quality balance, will try to match this pattern - however deploying Aave is not that trivial.

If deploying Aave pool from scratch takes too much time then you may test against a mocked instance. We'll refactor later.

> Couldn't our rpc manager help with the fail in requests problem?

Testing against a forked instance means testing against a predefined state. It's way better when you fully understand the protocol where you send assets to (i.e. when you deploy aave yourself) compared to forked tests where you use already deployed addresses of aave contracts, addresses of users who have funds, addresses of external protocols, etc...

Reply:
> I understand where you come from. This is a speed/quality balance, will try to match this pattern - however deploying Aave is not that trivial.

@zugdev  I think you may keep forked tests but we'll have to refactor them later. Can't we fork from mainnet instead of sepolia?

Reply:
I'd be down to refactor it later, but as you've said it will take time. Regarding protocol understanding, I fully understand how Aave operates, as I've forked it many times now. The main advantage of forking sepolia instead of mainnet is that we minimize the number of "magic addresses" we prank or use, paired with the permission to change protocol configs. Aave in sepolia is prepared to be tested on, they support mintable tokens for example. Another big advantage is interest rate and fees are set to very low, which allow us to assert that the deposit amount is equal to aToken balance post-deposit. 

It is up to you what chain we fork from, but I'd recommend Sepolia. You can further read why at https://docs.aave.com/developers/guides/testing-guide